lib/autoyast.rb

```diff
@@ -278,14 +284,38 @@ def apply_extracted_files(scope)
 
     @system_description[scope].files.map do |file|
       if file.user
-        @chroot_scripts << "chown #{file.user}:#{file.group} '#{File.join("/mnt", file.name)}'"
+        @chroot_scripts << "chown --no-dereference #{file.user}:#{file.group} '#{File.join("/mnt", file.name)}'"
       end
-      if file.mode
+      if file.mode && !file.link?
         @chroot_scripts << "chmod #{file.mode} '#{File.join("/mnt", file.name)}'"
       end
     end
   end
 
+  def apply_changed_managed_files
+    scope = "changed_managed_files"
+    return if !@system_description.scope_extracted?(scope)
+
+    @system_description[scope].files.each do |file|
+      if file.deleted?
+        @chroot_scripts << "rm -rf '#{file.name}'"
+      elsif file.directory?
+        @chroot_scripts << "chmod #{file.mode} '#{File.join("/mnt", file.name)}'"
+        @chroot_scripts << "chown #{file.user}:#{file.group} '#{File.join("/mnt", file.name)}'"
+      elsif file.file?
+        url = "`cat /tmp/description_url`/#{URI.escape(File.join(scope, file.name))}"
+        @chroot_scripts << "mkdir -p '#{File.join("/mnt", File.dirname(file.name))}'"
+        @chroot_scripts << "curl -o '#{File.join("/mnt", file.name)}' \"#{url}\""
+        @chroot_scripts << "chmod #{file.mode} '#{File.join("/mnt", file.name)}'"
+        @chroot_scripts << "chown #{file.user}:#{file.group} '#{File.join("/mnt", file.name)}'"
+      elsif file.link?
+        @chroot_scripts << "rm -rf '#{File.join("/mnt", file.name)}'"
+        @chroot_scripts << "ln -s '#{file.target}' '#{File.join("/mnt", file.name)}'"
+        @chroot_scripts << "chown --no-dereference #{file.user}:#{file.group} '#{File.join("/mnt", file.name)}'"
```

Comment:
Line is too long. [112/100]<br>Prefer single-quoted strings inside interpolations.